src/sid/msm.py

```diff
@@ -279,3 +281,11 @@ def _flatten_index(data):
 def _is_diagonal(mat):
     """Check if the matrix is diagonal."""
     return not np.count_nonzero(mat - np.diag(np.diagonal(mat)))
+
+
+def _is_dict_of_callables(candidate):
+    out = isinstance(candidate, dict)
+    for value in candidate.values():
+        if not callable(value):
+            out = False
+    return out
```

Comment:
Yours tested for dict and assumed a dict in the for loop.
```suggestion
def _is_dict_of_callables(x):
    return isinstance(x, dict) and all(callable(value) for value in x.values())
```